src/core/buffers/representation/representation_buffer.ts

```diff
@@ -369,6 +377,23 @@ export default function RepresentationBuffer<T>({
 
   return observableMerge(status$, bufferQueue$).pipe(share());
 
+  function validateContent(segment: ISegment) {
+    const { chunks, hasChunkComplete } = waitingSegmentCache[segment.id];
+    const hasAppendedAllChunks = !arrayFind(chunks, ([_, status]) => {
+      return status !== "appended";
```

Comment:
Why all this complexity?

When receiving "chunk-complete", either:
  - we're still appending the segment, in which case it become complete when done -> add a flag or something to the waiting segment queue
  - we're not appending the segment at this time, in which case we can validate the segment (even if it is has been canceled, we still want to validate that a part of the segment has been pushed)

We just have to make sure that the code reacting to "chunk-complete" is called after the code reacting to each data chunk. No?